src/main/java/org/codehaus/groovy/classgen/ClassCompletionVerifier.java

```diff
@@ -418,10 +418,6 @@ private void checkMethodsForIncorrectModifiers(final ClassNode cn) {
                 addError("The " + getDescription(method) + " from " + getDescription(cn) +
                         " must not be final. It is by definition abstract.", method);
             }
-            if (method.isStatic() && !method.isStaticConstructor()) {
-                addError("The " + getDescription(method) + " from " + getDescription(cn) +
-                        " must not be static. Only fields may be static in an interface.", method);
-            }
         }
     }
 
```

Comment:
`checkInterfaceMethodVisibility` may need some changes.  I rewrote it like this:
```java
    private void checkInterfaceMethodVisibility(final ClassNode node) {
        if (!node.isInterface()) return;
        for (MethodNode method : node.getMethods()) {
            if (!method.isPublic()) {
                if (method.isAbstract()) {
                    addError("Method '" + method.getName() + "' from " + getDescription(node) +
                            " must be public as it is declared as an abstract method.", method);
                } else if (!method.isPrivate() && !method.isStaticConstructor()) {
                    addError("Method '" + method.getName() + "' is " +
                            (method.isProtected() ? "protected" : "package-private") +
                            " but should be private in " + getDescription(node) + ".", method);
                }
            }
        }
    }
```